helper/openssl_certificate_loader.go

```diff
@@ -21,26 +21,65 @@ import (
 	"os"
 
 	"github.com/paketo-buildpacks/libpak/bard"
+	"github.com/paketo-buildpacks/libpak/sherpa"
+	"golang.org/x/sys/unix"
 
 	"github.com/paketo-buildpacks/libjvm"
 )
 
+const TmpTrustStore = "/tmp/truststore"
+
 type OpenSSLCertificateLoader struct {
 	CertificateLoader libjvm.CertificateLoader
 	Logger            bard.Logger
 }
 
+func (o OpenSSLCertificateLoader) prepareTempTrustStore(trustStore, tempTrustStore string) (map[string]string, error) {
+	o.Logger.Infof("Using readonly truststore: %s", tempTrustStore)
+
+	trustStoreFile, err := os.Open(trustStore)
+	if err != nil {
+		return nil, fmt.Errorf("unable to open trust store %s\n%w", trustStore, err)
+	}
+	defer trustStoreFile.Close()
+
+	err = sherpa.CopyFile(trustStoreFile, tempTrustStore)
+	if err != nil {
+		return nil, fmt.Errorf("failed to copy dir (%s, %s)\n%w", trustStore, tempTrustStore, err)
+	}
+
+	opts := sherpa.AppendToEnvVar("JAVA_TOOL_OPTIONS", " ", fmt.Sprintf("-Djavax.net.ssl.trustStore=%s", tempTrustStore))
```

Comment:
What do you think about checking to see if this setting already exists?

My only hesitation is that I'm not sure we could do anything, aside from log a warning message and continue on.

Thoughts?

Reply:
> Thoughts?

We discussed this and would probably rather not check this. For 2 reasons:

1.) What should we do in that case? Import into that one? Probably we should fail in that case.
2.) What would be the reason to have a custom truststore? If you want to bring your custom `cacerts` at runtime, you could do so with a binding.

I think it is too much of an edge case to add some extra logic for this.